### PR TITLE
Add support for Doc2Vec in the Phrases class

### PR DESCRIPTION
**Pull request by tmthyjames**
Previous to this commit, I couldn't easily get phrases to play nice with Doc2Vec (I wanted bigrams + document vectors). I hope this solves that issue without breaking other things! 

Please let me know if this doesn't follow best practices. Thanks! Great tool!

### CODE REVIEW

File: gensim/models/phrases.py [hunk 1]
```diff
@@ -3,15 +3,11 @@
 # Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
 
 """Automatically detect common phrases -- multi-word expressions / word n-grams -- from a stream of sentences.
-
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: please revert empty lines (here and everywhere)

File: gensim/models/phrases.py [hunk 2]
```diff
@@ -425,6 +396,10 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000,
                     "PROGRESS: at sentence #%i, processed %i words and %i word types",
                     sentence_no, total_words, len(vocab),
                 )
+            
+            if doc2vec:
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: What for? Please show several code pieces that demonstrate your idea (why this is needed)

File: gensim/models/phrases.py [hunk 3]
```diff
@@ -237,9 +220,8 @@ class Phrases(SentenceAnalyzer, PhrasesTransformation):
 
     def __init__(self, sentences=None, min_count=5, threshold=10.0,
                  max_vocab_size=40000000, delimiter=b'_', progress_per=10000,
-                 scoring='default', common_terms=frozenset()):
+                 scoring='default', common_terms=frozenset(), doc2vec=False):
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: In general, this idea is bad: Phrases shouldn't depends on `doc2vec` or any other model

File: gensim/models/phrases.py [hunk 4]
```diff
@@ -425,6 +396,10 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000,
                     "PROGRESS: at sentence #%i, processed %i words and %i word types",
                     sentence_no, total_words, len(vocab),
                 )
+            
+            if doc2vec:
+              sentence = sentence.words
```
Comment (menshikh-iv) [inline gensim/models/phrases.py]: Probably here you "unapack" `TaggedDocument` class, you should do it before you pass it to phrases, no additional code needed here, something like

```python
sentences_for_d2v =  # list of TaggedDocument

sentences_for_phrases = [s.words for s in sentences]
phr = Phrases(sentences_for_phrases)

```

**Comment by menshikh-iv** [comment]
@tmthyjames I don't think that this is useful change (see my example from the review), why you need this?

**Comment by menshikh-iv** [comment]
PR abandoned (+ doesn't looks useful), I close it.

**Comment by Anushree6** [comment]
This integration of Phrases and  doc2vec could have been so useful. This totally makes sense to me and shouldn't have been discouraged and closed.

**Comment by menshikh-iv** [comment]
hi @Anushree6, 

>  This totally makes sense to me and shouldn't have been discouraged and closed.

How will you plan to use that? Feel free to suggest several code examples.

I do not understand whats profit here, please explain it to me.

**Pull request closed without merging**